corehq/apps/reports/tasks.py

```diff
@@ -158,9 +160,56 @@ def apps_update_calculated_properties():
         send_to_elasticsearch('apps', doc, es_merge_update=True)
 
 
-@task(serializer='pickle', ignore_result=True)
-def export_all_rows_task(ReportClass, report_state, recipient_list=None, subject=None):
-    report = object.__new__(ReportClass)
+@task(ignore_result=True)
+def export_all_rows_task(report_class, report_state, recipient_list=None, subject=None):
+    #will move imports to the top
+    from corehq.apps.reports.standard.monitoring import (
+        WorkerActivityReport,
+        DailyFormStatsReport,
+        CaseActivityReport,
+    )
+    from corehq.apps.reports.standard.deployments import ApplicationStatusReport
+    from phonelog.reports import DeviceLogDetailsReport
+    from corehq.apps.reports.standard.cases.case_list_explorer import CaseListExplorer
+    from corehq.apps.reports.standard.cases.duplicate_cases import DuplicateCasesExplorer
+    from corehq.apps.reports.commtrack import CurrentStockStatusReport
+    from corehq.apps.smsbillables.interface import SMSBillablesInterface, SMSGatewayFeeCriteriaInterface
+    from corehq.apps.enterprise.interface import EnterpriseSMSBillablesReport
+    from corehq.apps.hqadmin.reports import DeviceLogSoftAssertReport
+
+    #for testing
+    temp_dict = {
+        ApplicationStatusReport.slug: ApplicationStatusReport,
+        CaseActivityReport.slug: CaseActivityReport,
+        DailyFormStatsReport.slug: DailyFormStatsReport,
+        DeviceLogDetailsReport.slug: DeviceLogDetailsReport,
+        SMSBillablesInterface.slug: SMSBillablesInterface,
+        SMSGatewayFeeCriteriaInterface.slug: SMSGatewayFeeCriteriaInterface,
+        WorkerActivityReport.slug: WorkerActivityReport,
+        DeviceLogSoftAssertReport.slug: DeviceLogSoftAssertReport,
+        EnterpriseSMSBillablesReport.slug: EnterpriseSMSBillablesReport,
+        CaseListExplorer.slug: CaseListExplorer,
+        DuplicateCasesExplorer.slug: DuplicateCasesExplorer,
+        CurrentStockStatusReport.slug: CurrentStockStatusReport,
+    }
+
+    #Reforming datespan object
+    if 'startdate' and 'enddate' in report_state['request_params']:
+        start_date = datetime.strptime(str(report_state['request_params']['startdate']), '%Y-%m-%d %H:%M:%S')
+        end_date = datetime.strptime(str(report_state['request_params']['enddate']), '%Y-%m-%d %H:%M:%S')
```

Comment:
Since you should be using `.isoformat()` when serializing datetimes, you can use `iso_string_to_datetime` to convert back to datetimes.